portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles comments data */
+@WebServlet("/comments")
+public class CommentsServlet extends HttpServlet {
+
+  private ArrayList<String> comments;
+  private ArrayList<String> names;
+  private ArrayList<ArrayList<String>> namesAndComments;
+  private ArrayList<String> labels;
+
+  
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+    names = new ArrayList<>();
+    namesAndComments = new ArrayList<ArrayList<String>>();
+    labels = new ArrayList<>();
+  }
+  
+  /* responds with a JSON string containing names and comments*/
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Convert the ArrayLists to JSON
+    namesAndComments.add(names);
+    namesAndComments.add(comments);
+    labels.add("names");
+    labels.add("comments");
+    String namesAndCommentsListJson = convertListToJson(namesAndComments, labels);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(namesAndCommentsListJson);
+  }
+
+  /* when the submit button on the comments page is hit, doPost requests the text in the name and commment field and stores them*/ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the comments form
+    String nameText = getParameter(request, "name-text", "");
+    String commentText = getParameter(request, "comment-text", "");
+    names.add(nameText);
+    comments.add(commentText);
+
+    // redirect back to comments page
+    response.sendRedirect("/comments-page.html");
+  }
+   
+   /* converts a list of objects and a list of names into a JSON string */
```

Comment:
I imagine both of these must exist aas java libraries. Can you if you can find one that does what you want so that you don't need to build them yourself?

Reply:
Looks like GSON is a google provided API that does this, I will change.

Reply:
To implement this change, I created a new Comment java class and created Comment objects which the library turned into a JSON string.